core/org.eclipse.cdt.core.native/src/org/eclipse/cdt/utils/pty/PTY2.java

```diff
@@ -0,0 +1,69 @@
+/********************************************************************************
+ * Copyright (c) 2022 徐持恒 Xu Chiheng
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ********************************************************************************/
+package org.eclipse.cdt.utils.pty;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import org.eclipse.cdt.utils.spawner.ProcessFactory;
+import org.eclipse.core.runtime.Platform;
+
+public class PTY2 {
+	private Process terminalEmulator;
+	private String slaveName;
+
+	public PTY2() throws IOException {
+		// launch the patched terminal emulator process, and get the pty's slave name from stdout of the
+		// terminal emulator process
+		String command = null;
+		InputStream in = null;
+		try {
+			command = PTY2Util.getTerminalEmulatorCommand();
+			// both mintty.exe and konsole support --opentty and --title option
+			String[] cmdArray;
+			if (Platform.getOS().equals(Platform.OS_WIN32)) {
+				cmdArray = new String[] { command, "--openpty", "--hold=always", //$NON-NLS-1$ //$NON-NLS-2$
+						"--title", "Terminal Emulator" }; //$NON-NLS-1$ //$NON-NLS-2$
+			} else if (Platform.getOS().equals(Platform.OS_LINUX)) {
+				cmdArray = new String[] { command, "--openpty", "--nofork", "--hold", //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
```

Comment:
We're not going to merge code into CDT that depends on mintty/konsole adding command line support for openpty until they indicate a likelihood of accepting it. At the moment they have indicated an unwillingness to add --openpty.

[xrefs](https://github.com/eclipse-cdt/cdt/pull/193#issuecomment-1338109747):

[mintty/mintty#1187 (comment)](https://github.com/mintty/mintty/issues/1187#issuecomment-1336200885)
https://bugs.kde.org/show_bug.cgi?id=462495
https://gitlab.gnome.org/GNOME/gnome-terminal/-/issues/7951